### PR TITLE
Collect missing tokens in the language server

### PR DESCRIPTION
**Pull request by JoshuaBatty**
This PR collects most of the remaining tokens in the language server. 

I wrote a script that saved out the ASTs for the `sway_ast::Module`, `ParseProgram` and `TypedProgram` stages for each of the sway examples and all of the e2e/language/should pass examples. This made it quick to inspect what extra span information needed to be carried through in the compiler.

There are still a few tokens not being collected but they don't involve trivial changes to the compiler so I'll address them individually in separate PRs.

### CODE REVIEW

File: sway-lsp/src/core/traverse_parse_tree.rs
Comment (mitchmindtree) [inline sway-lsp/src/core/traverse_parse_tree.rs:621]: Do we have an issue open? Should link it here.

**Review by mitchmindtree — APPROVED** [review]
UtACK, looking good!

One minor nit but will leave that up to you, feel free to ping me for re-approve if you want to address it.

File: sway-lsp/src/core/traverse_parse_tree.rs
Comment (JoshuaBatty) [reply sway-lsp/src/core/traverse_parse_tree.rs]: Just added a link to the issue.

**Review by mitchmindtree — APPROVED** [review]

**Review by mohammadfawaz — APPROVED** [review]
Seems reasonable!

**Pull request merged**
